pkgs/by-name/cu/cup-docker/package.nix

```diff
@@ -8,56 +8,66 @@
   nix-update-script,
   withServer ? true,
 }:
-let
+
+rustPlatform.buildRustPackage (finalAttrs: {
   pname = "cup-docker";
   version = "3.4.0";
+
   src = fetchFromGitHub {
     owner = "sergi0g";
     repo = "cup";
-    tag = "v${version}";
+    tag = "v${finalAttrs.version}";
     hash = "sha256-ciH3t2L2eJhk1+JXTqEJSuHve8OuVod7AuQ3iFWmKRE=";
   };
-  web = stdenvNoCC.mkDerivation (finalAttrs: {
-    pname = "${pname}-web";
-    inherit version src;
+
+  web = stdenvNoCC.mkDerivation{
```

Comment:
I don't really agree with this since this would cause the main derivation to always be dependent on the web derivation. IE rebuilds. I think you may need to add something to make this conditional.